database/index.js

```diff
@@ -1,76 +1,99 @@
-const mysql = require('promise-mysql')
-const legacy = require('./legacy.js')
+const legacy = require("./legacy.js");
 var Promise = require("bluebird");
-var getSqlConnection = require('./dbConnection.js');
+var getMySqlConnection = require("./mysqlConnection");
+// var getMongoConnection = require("./mongoConnection"); // TODO = allow configurable mongo use
 
 // used by client
-const getListingData = (id) => {
-	return legacy.getListingData(id);
-}
+const getListingData = id => {
+  return legacy.getListingData(id);
+};
 
 const getListings = () => {
-	console.log('db get listings')
-	return Promise.using(getSqlConnection(), function(connection) {
-    return connection.query('select * from apartment').then(function(rows) {
-      return (rows)
-    }).catch(function(error) {
-      console.log(error);
-    });
-  })
-}
+  console.log("db get listings");
+  return Promise.using(getMySqlConnection(), function(connection) {
```

Comment:
I like your use of bluebird for establishing a promise-based sql connection. Are there other benefits as well? I'm wondering because Promises are native to nodejs, so you could accomplish the same without the dependency.

Reply:
Thanks! Bluebird is being used for it's library function 'disposer,' which is used to release each mysql connection.  I'm using 'promise-mysql' which includes Bluebird, although a refactor to eliminate that dependency and lean on Node's native promises would be worthwhile.

I'm having to duplicate this implementation for my mongo connection, re: dependency injection as you mentioned, which could force my hand on the refactor